docs/backend-system/architecture/07-naming-patterns.md

```diff
@@ -0,0 +1,94 @@
+---
+id: naming-patterns
+title: Backend System Naming Patterns
+sidebar_label: Naming Patterns
+# prettier-ignore
+description: Naming patterns in the backend system
+---
+
+These are the naming patterns to adhere to withing the backend system. They help us keep exports consistent across packages and make it easier to understand the usage and intent of exports.
+
+### Plugins
+
+| Description | Pattern      | Examples                            |
+| ----------- | ------------ | ----------------------------------- |
+| export      | `<id>Plugin` | `catalogPlugin`, `scaffolderPlugin` |
+| ID          | `'<id>'`     | `'catalog'`, `'scaffolder'`         |
+
+Example:
+
+```ts
+export const catalogPlugin = createBackendPlugin({
+  id: 'catalog',
+  ...
+})
+```
+
+### Modules
+
+| Description | Pattern                      | Examples                            |
+| ----------- | ---------------------------- | ----------------------------------- |
+| export      | `<pluginId>Module<ModuleId>` | `catalogModuleGithubEntityProvider` |
```

Comment:
This is not our current naming pattern, thoughts on this change @backstage/maintainers?

The main reason I'm proposing this change is to have a better grouping at at-glance information about what modules are installed for each plugin in the backend index file. For example, compare our existing pattern:

```ts
backend.add(catalogPlugin());
backend.add(githubEntityProviderCatalogModule());
backend.add(scaffolderCatalogModule());
backend.add(scaffolderPlugin());
```

I feel it's quite hard to quickly glance what modules are installed for each plugin, and it would require additional grouping and/or comments for it to be clear. It's not even immediately obvious which features are modules and which are plugins. You kind of need to know to bounce back and forth between the beginning and the end of the lines when reading.

To me, this alternative naming pattern feels a lot more quick to digest:

```ts
backend.add(catalogPlugin());
backend.add(catalogModuleGithubEntityProvider());
backend.add(catalogModuleScaffolder());
backend.add(scaffolderPlugin());
```

Reply:
+1 from me.

Reply:
same here